### PR TITLE
IM Component

### PR DESCRIPTION
**Pull request by joelridden**
### **IM Components**

Adds the functionality to select a given IM Component when selecting an IM in both Hazard and Project.
image
image

- Changed drop down selects to be separate between a IMCustomSelect and a standard CustomSelect (to reduce complexity)
- IM Component is now sent on each request
- Fetch of IM's now is a dictionary which includes the IM's supported components and periods if needed

### CODE REVIEW

File: frontend/src/components/Hazard/SeismicHazard/HazardViewerUHS.js [hunk 1]
```diff
@@ -105,6 +106,7 @@ const HazardViewerUHS = () => {
             station: station,
             exceedances: `${getExceedances().join(",")}`,
             calc_percentiles: 1,
+            im_component: selectedIMComponent === null ? "RotD50" : selectedIMComponent,
```
Comment (SnowNooDLe) [inline frontend/src/components/Hazard/SeismicHazard/HazardViewerUHS.js:109]: Is `im_component` default to **RotD50** only for UHS?

Comment (joelridden) [reply frontend/src/components/Hazard/SeismicHazard/HazardViewerUHS.js:109]: Yes as that's the only one you can press compute for without setting the IM / Period / IM Component dropdowns

**Review by SnowNooDLe — APPROVED** [review]
Happy to have a split CustomSelect, one for IM another one for Non-IM.
IMCustomSelect's placeholder logic seems longer/bigger than I expected but still, we now only have to care about IM with IMCustomSelect, not every dropdown.

You probably just need to fix merge conflicts.

**Review by claudio525 — APPROVED** [review]

**Review by SnowNooDLe — DISMISSED** [review]

File: frontend/src/components/common/CustomSelect.js [hunk 2]
```diff
@@ -54,17 +52,6 @@ const CustomSelect = ({
...
-  useEffect(() => {
-    if (selectedIM !== null && selectedIM["value"] !== "pSA") {
-      setSelect(null);
-      selectInputRef.current.select.clearValue();
-    }
-  }, [selectedIM]);
-
   /* 
     Non-projects
     By user, the site got chagned by clicking setting location
```
Comment (jasonmotha) [inline frontend/src/components/common/CustomSelect.js:57]: changed* Even though it's not your comment, it'll be good to change.

File: frontend/src/components/common/IMCustomSelect.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  useEffect(() => {
+    if (selectedIM !== null && selectedIM["value"] !== "pSA") {
+      setSelect(null);
+      selectInputRef.current.select.clearValue();
+    }
+  }, [selectedIM]);
+
+  /* 
+    Non-projects
+    By user, the site got chagned by clicking setting location
```
Comment (jasonmotha) [inline frontend/src/components/common/IMCustomSelect.js:59]: Same speeling error

**Review by SnowNooDLe — DISMISSED** [review]

**Review by jasonmotha — APPROVED** [review]

**Review by claudio525 — APPROVED** [review]

**Comment by claudio525** [comment]
I fixed the Changelog conflict. Since everyone had approved before that change I'm going to commit this.

**Pull request merged**